esp/services/ws_sql/SQL2ECL/HPCCFile.cpp

```diff
@@ -127,6 +127,39 @@ bool HPCCFile::getFileRecDefwithIndexpos(HPCCColumnMetaData * fieldMetaData, Str
     return false;
 }
 
+bool setChildColumns(HPCCColumnMetaData * parent, IPropertyTree * fieldtree)
+{
+    StringBuffer ecltype;
+    StringBuffer colname;
+    int colsize;
+    int colindex;
+
+    if (parent == nullptr || fieldtree == nullptr)
+    {
+        ESPLOG(LogMin, "Could not set HPCC file childcolumns!");
+        return false;
+    }
+
+    Owned<IPropertyTreeIterator> fields = fieldtree->getElements("Field");
+    ForEach(*fields)
+    {
+        IPropertyTree & curField = fields->query();
+        curField.getProp("@ecltype", ecltype.clear());
+        curField.getProp("@name", colname.clear());
+
+        Owned<HPCCColumnMetaData> col = HPCCColumnMetaData::createHPCCColumnMetaData(colname.str());
+        col->setIndex(curField.getPropInt("@position", -1));
+
+        if (strncmp(ecltype, "table of", 8)==0)
```

Comment:
@rpastrana Is ecltype a value we are setting or is this coming from the users SQL? Should using regex rather than strncmp in these instances?

Reply:
It would be much better to test against a type code, but I think we're ok w/ this string-wise comparison since the @ecltype attribute is processed from the RTLType info.